test/components/bitvector.cpp

```diff
@@ -0,0 +1,208 @@
+                            40,   63,   64,   65,    127,   128, 129,
+                            1000, 1024, 2000, 10000, 100000}
+    {}
+
+    gko::array<index_type> create_random_values(index_type num_values,
+                                                index_type size)
+    {
+        std::vector<index_type> values(num_values);
+        std::uniform_int_distribution<index_type> dist(
+            0, std::max(size - 1, index_type{}));
+        for (auto& value : values) {
+            value = dist(this->rng);
+        }
+        std::sort(values.begin(), values.end());
+        values.erase(std::unique(values.begin(), values.end()), values.end());
+        return gko::array<index_type>{this->ref, values.begin(), values.end()};
+    }
+
+    void assert_bitvector_equal(const bitvector& bv, const bitvector& dbv)
+    {
+        ASSERT_EQ(bv.get_size(), dbv.get_size());
+        const auto num_blocks =
+            static_cast<gko::size_type>(bv.get_num_blocks());
+        const auto bits =
+            gko::detail::array_const_cast(gko::make_const_array_view(
+                bv.get_executor(), num_blocks, bv.get_bits()));
+        const auto dbits =
+            gko::detail::array_const_cast(gko::make_const_array_view(
+                dbv.get_executor(), num_blocks, dbv.get_bits()));
+        const auto ranks =
+            gko::detail::array_const_cast(gko::make_const_array_view(
+                bv.get_executor(), num_blocks, bv.get_ranks()));
+        const auto dranks =
+            gko::detail::array_const_cast(gko::make_const_array_view(
+                dbv.get_executor(), num_blocks, dbv.get_ranks()));
+        GKO_ASSERT_ARRAY_EQ(bits, dbits);
+        GKO_ASSERT_ARRAY_EQ(ranks, dranks);
+    }
+
+    std::default_random_engine rng;
+    std::vector<index_type> sizes;
+};
+
+TYPED_TEST_SUITE(Bitvector, gko::test::IndexTypes, TypenameNameGenerator);
+
+
+TYPED_TEST(Bitvector, BuildFromIndicesIsEquivalentToRef)
+{
+    using index_type = typename TestFixture::index_type;
+    using bitvector = typename TestFixture::bitvector;
+    for (auto size : this->sizes) {
+        SCOPED_TRACE(size);
+        for (auto num_values :
+             {index_type{}, size / 10, size / 4, size / 2, size}) {
+            SCOPED_TRACE(num_values);
+            auto values = this->create_random_values(num_values, size);
+            gko::array<index_type> dvalues{this->exec, values};
+
+            auto bv = gko::kernels::reference::bitvector::from_sorted_indices(
+                this->ref, values.get_data(), values.get_size(), size);
+            auto dbv = gko::kernels::GKO_DEVICE_NAMESPACE::bitvector::
+                from_sorted_indices(this->exec, dvalues.get_data(),
+                                    dvalues.get_size(), size);
+
+            this->assert_bitvector_equal(bv, dbv);
+        }
+    }
+}
+
+
+template <typename IndexType>
+std::pair<gko::bitvector<IndexType>, gko::bitvector<IndexType>> run_predicate(
```

Comment:
why is this its own function? It's only used once, so I don't see any benefit.

Reply:
Same as in the other test, nvcc has issues with device lambdas defined inside template member functions

Reply:
:'-(